mockzilla/src/commonMain/kotlin/com/apadmi/mockzilla/lib/internal/utils/SocketIo.kt

```diff
@@ -0,0 +1,11 @@
+package com.apadmi.mockzilla.lib.internal.utils
+
+import io.ktor.network.selector.SelectorManager
+import io.ktor.network.sockets.aSocket
+
+internal class SocketIo(private val selectorManager: SelectorManager) {
+    suspend fun isPortAvailable(port: Int): Boolean = runCatching {
+        val serverSocket = aSocket(selectorManager).tcp().bind("127.0.0.1", port)
+        serverSocket.dispose()
+    }.isSuccess
```

Comment:
If given 0 as the port ktor will automatically bind it, not sure if that'll break this code? Might be worth handling 0 as a special case?

Reply:
@SamC-Apadmi Didn't realise, what would be expected in this special case? We allow Mockzilla to start or fail?

Reply:
Upon reviewing the docs, I see that ktor doesn't bind _to port 0_, it merely just treats port 0 as a special flag to select a random available port. I think in this case, it would be expected that we simply skip the check [as it is invalid] and allow the server to start.